### PR TITLE
test(core): cover expressions.ts uncovered paths (Phase 2)

### PR DESCRIPTION
**Pull request by hawkeyexl**
## Phase 2 of: root coverage ratchet

Builds on the Phase 1 baseline (#420). Adds `test/expressions-coverage.test.js` (55 cases) covering the previously-uncovered code in `src/core/expressions.ts`, driven entirely through its 4 public exports.

### Coverage impact (isolated c8 on expressions.ts)
| Metric | Before | After |
|---|---|---|
| Lines | 86.6% | **98.3%** |
| Branches | 68.5% | **95.2%** |
| Functions | 100% | 100% |

The 4 lines that remain uncovered are genuinely unreachable through the public API (inner non-string guards / error arms that the outer `resolveExpression` already handles before delegating). Left untouched rather than adding `c8-ignore` comments to source.

### What's covered
Embedded `{{...}}` resolution, `jq` evaluation, operator/object/error paths, `getMetaValue` JSON-pointer + array-index edge cases, and the comparison-operand quoting/coercion helpers.

### Process note
- **Test-only — no source changes.** All 112 expressions tests pass (57 existing + 55 new).
- Thresholds in `coverage-thresholds.json` are **not bumped in this commit**. The blocking `Coverage ratchet (root)` job measures the new full-suite global coverage; I'll then commit the raised baseline (the bump this phase earned) so the gain is locked in before merge — same two-step pattern as the Phase 1 baseline capture.

Next phases (separate PRs): `src/core/utils.ts` (91%), `src/utils.ts` (70%), then reporters/hints/etc.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded automated coverage for expression resolution, meta value replacement, meta value lookup, and assertion evaluation.
  * Added checks for embedded `{{...}}` rendering, JSON-stringification for object results, `String()` conversion for scalar embedded values, and deterministic fallback when embedded `jq(...)` fails.
  * Covered JSON pointer traversal (including missing segments and mixed array/object access), `{{var}}` template resolution with unknowns, and robust quoting/boolean-coercion rules in comparisons and operators.
  * Validated operator helper behavior (`contains`, `matches`, `oneOf`, `extract`) including edge cases like invalid regex and non-array `oneOf` inputs, plus outer error/catch behavior.
* **Chores**
  * Increased the branch coverage threshold baseline.

### CODE REVIEW

File: test/expressions-coverage.test.js [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  // Instead, exercise a jq() inside {{...}} with invalid jq: the inner
+  // resolveExpression catches it and returns the original, so the outer string
+  // still renders deterministically (covers the object/scalar paths around it).
+  it("an embedded jq with invalid syntax renders without crashing", async function () {
+    const r = await resolveExpression({
+      expression: "r={{jq($$d, @@@invalid)}}",
+      context: { d: { a: 1 } },
+    });
+    assert.equal(typeof r, "string");
+    assert.ok(r.startsWith("r="));
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/expressions-coverage.test.js]: Fixed — quoted the jq query (`jq($$d, "@@@invalid")`) so the inner expression is valid JS exercising jq's error path rather than a JS SyntaxError, and replaced the weak `startsWith` with an exact assertion on the rendered output. Also corrected the comment.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: coverage-thresholds.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "description": "Coverage baseline thresholds for the root doc-detective package, measured from the full root test suite (test/*.test.js) under c8 in CI. These values should only increase, never decrease.",
   "lastUpdated": "2026-06-30",
-  "lines": 82.25,
-  "statements": 82.25,
+  "lines": 82.45,
+  "statements": 82.45,
   "functions": 85.94,
-  "branches": 79.95
+  "branches": 80.54
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply coverage-thresholds.json:7]: Updated the PR description to match. The bump is intentionally part of this PR: it's the measure-then-commit completion of the per-phase loop — the Coverage ratchet (root) job measured the new full-suite coverage and that exact number (lines/statements 82.45%, branches 80.54%) is committed here so the gain is locked in before merge. The only non-test file touched is coverage-thresholds.json.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/expressions-coverage.test.js [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+  it("an embedded jq with an invalid query renders the sub-expression verbatim", async function () {
+    const r = await resolveExpression({
+      expression: 'r={{jq($$d, "@@@invalid")}}',
+      context: { d: { a: 1 } },
+    });
+    assert.equal(r, 'r=jq($$d, "@@@invalid")');
+  });
+});
+
+describe("expressions coverage: evaluateExpression operator helpers", function () {
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/expressions-coverage.test.js]: Renamed — the block is now 'operator helpers via the public API (contains/matches/oneOf/extract)' since the tests drive evaluateAssertion/resolveExpression, not the internal evaluateExpression.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
